packages/compass-app-stores/src/stores/instance-store.js

```diff
@@ -251,6 +251,17 @@ store.onActivated = (appRegistry) => {
     store.refreshInstance(appRegistry);
   });
 
+  appRegistry.on('collection-renamed', async ({ from, to }) => {
+    // we must fetch the old collection's metadata before refreshing because refreshing the
+    // collection metadata will remove the old collection from the model.
+    const metadata = await store.fetchCollectionMetadata(from);
```

Comment:
q: what happens if this one fails? do we have some magic catch all handling on the `appRegistry` or the rejection would fly around?